python/pyarrow/scalar.pxi

```diff
@@ -297,42 +297,73 @@ cdef dict _DATETIME_CONVERSION_FUNCTIONS = {}
 cdef c_bool _datetime_conversion_initialized = False
 
 
+cdef _add_micros_maybe_localize(dt, micros, tzinfo):
+    import pytz
+    if tzinfo is not None:
+        if not isinstance(tzinfo, datetime.tzinfo):
+            tzinfo = string_to_tzinfo(tzinfo)
+        dt = dt.replace(microsecond=micros, tzinfo=tzinfo)
```

Comment:
I don't think you need to replace the tzinfo here, since you are converting to it on the line below? 

From a quick test it does not seem to matter:

```
In [66]: d = datetime.datetime(2012,1,1,9,0,0)

In [67]: tzinfo = pytz.timezone('US/Eastern') 

In [68]: d2 = d.replace(tzinfo=tzinfo) 

In [69]: d2                  
Out[69]: datetime.datetime(2012, 1, 1, 9, 0, tzinfo=<DstTzInfo 'US/Eastern' LMT-1 day, 19:04:00 STD>)

In [70]: tzinfo.fromutc(d2)        
Out[70]: datetime.datetime(2012, 1, 1, 4, 0, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>)

In [71]: tzinfo.fromutc(d)   
Out[71]: datetime.datetime(2012, 1, 1, 4, 0, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>)
```

Reply:
Fixing